src/wstool/multiproject_cli.py

```diff
@@ -621,6 +623,59 @@ def cmd_diff(self, target_path, argv, config=None):
 
         return False
 
+    def cmd_foreach(self, target_path, argv, config=None):
+        """Run shell commands in each repository."""
+        parser = argparse.ArgumentParser(
+            usage='usage: %s foreach [command] [OPTIONS]' % self.progname,
+            description=__MULTIPRO_CMD_DICT__['foreach'],
+            epilog='See: http://www.ros.org/wiki/rosinstall for details')
+        parser.add_argument('command', type=str, help='command to run')
+        parser.add_argument('--shell', default=False,
+                            help='use the shell as the program to execute',
+                            action='store_true')
+        parser.add_argument('--no-stdout', dest='show_stdout',
+                            default=True,
+                            help='do not show stdout',
+                            action='store_false')
+        parser.add_argument('--no-stderr', dest='show_stderr',
+                            default=True,
+                            help='do not show stderr',
+                            action='store_false')
+        # -t option required here for help but used one layer above
+        # see cli_common
+        parser.add_argument("-t", "--target-workspace", dest="workspace",
+                            default=None,
+                            help="which workspace to use",
+                            action="store")
+        args = parser.parse_args(argv)
+
+        if config is None:
+            config = multiproject_cmd.get_config(
+                target_path,
+                additional_uris=[],
+                config_filename=self.config_filename)
+        elif config.get_base_path() != target_path:
+            raise MultiProjectException('Config path does not match %s %s' %
+                                        (config.get_base_path(), target_path))
+
+        # run shell command
+        outputs = multiproject_cmd.cmd_foreach(config,
+                                               command=args.command,
+                                               shell=args.shell)
+        for output in outputs:
+            localname = output['entry']._local_name
+            if args.show_stdout:
+                for line in output['stdout'].split('\n'):
+                    if not line.strip():
+                        continue
+                    sys.stdout.write('[%s] %s\n'% (localname, line))
+            if args.show_stderr:
+                for line in output['stderr'].split('\n'):
+                    if not line.strip():
```

Comment:
not sure if it is a good idea to drop empty lines, they can be meaningful output in this context.